### PR TITLE
CUDOS-1345: Create verify plugin

### PR DESCRIPTION
**Pull request by aemil145**
- Implement verify plugin - extends environment and creates a custom task
- updated custom tasks
- fixed e2e tests
- small linting rules updates

### CODE REVIEW

File: packages/blast-verify/verify.js [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+  // parameters validation
+  if (!localContractLabel) {
+    throw new BlastVerifyError('Invalid contract label!')
+  }
+  if (!fs.existsSync(path.join(getProjectRootPath(), 'contracts', localContractLabel))) {
+    throw new BlastVerifyError(`"${localContractLabel}" contract folder not found! ` +
+      'Make sure your contract is in /contracts folder.')
+  }
+  if (!contractAddress) {
+    throw new BlastVerifyError('Invalid contract address!')
```
Comment (maptuhec) [inline packages/blast-verify/verify.js]: invalid is not the best error here imo, 'missing contract address' maybe? becuase the user is actualy lacking an address. The validation of the address is in the next function

File: packages/blast-verify/verify.js [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+    if (data.verificationError && data.verificationError.length > 0) {
+      console.log(`Verification failed! Reason: ${data.verificationError}`)
+      return data
+    }
+    if (data.parsingError && data.parsingError.length > 0) {
+      console.log(`Parsing failed! Reason: ${data.parsingError}`)
+      return data
+    }
+    // watch for the progress depending on API response
+    if (!isVerified && data.verified === true) {
```
Comment (maptuhec) [inline packages/blast-verify/verify.js]: are you sure that `isVerified` needs to be false?

File: packages/blast-verify/verify.js [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+        }
+      })
+      .pipe(outputStream)
+    outputStream.on('close', () => { resolve() })
+    outputStream.on('end', () => { reject(new BlastVerifyError('Archiver error: Data has been drained')) })
+    archive.finalize()
+  })
+}
+
+const isValidAddress = (address, addressPrefix) => {
```
Comment (maptuhec) [inline packages/blast-verify/verify.js]: this function can be transferred in a utilities folder

File: packages/blast-verify/verify.js [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+
+  const id = verificationResponse.data.id
+  console.log('Verification job is submitted. Getting the result may take a long time. Now you can opt to safely free' +
+    ' your terminal by killing the process (ctrl+C) and check the status manually on ' +
+    `${apiUrl}/verification-status?id=${id}`)
+
+  // Periodically check if the verification job is done
+  console.log('Verifying contract...')
+  let isVerified = false
+  while (1) {
```
Comment (maptuhec) [inline packages/blast-verify/verify.js]: is this the best solution? hardcoded statemnts are a bit odd

File: packages/blast-verify/verify.js [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+  if (!networks[config.verify.network]) {
+    throw new BlastVerifyError('Invalid network passed in the config file! Use "blast verify --ls" to show ' +
+      'all available networks.')
+  }
+  return networks[config.verify.network]
+}
+
+const createContractsArchive = (outputArchiveDir) => {
+  const contractsDir = path.join(getProjectRootPath(), 'contracts')
+  const packagesDir = path.join(getProjectRootPath(), 'packages')
```
Comment (maptuhec) [inline packages/blast-verify/verify.js]: why aren't you checking for packagesDIr?

**Review by maptuhec — CHANGES_REQUESTED** [review]

File: packages/blast-verify/verify.js (hunk 1 above)
Comment (aemil145) [reply packages/blast-verify/verify.js]: fixed

File: packages/blast-verify/verify.js (hunk 2 above)
Comment (aemil145) [reply packages/blast-verify/verify.js]: The aim is to show the message once out of all periodic checks and I'm using `isVerified` to achieve this. It should be false --> show the message --> assign to true --> don't show again

File: packages/blast-verify/verify.js (hunk 5 above)
Comment (aemil145) [reply packages/blast-verify/verify.js]: because in case of contracts that do not import anything, `packages` folder is not required

**Review by maptuhec — APPROVED** [review]

**Pull request merged**